iothub/service/tests/Registry/DevicesClientTests.cs

```diff
@@ -558,5 +689,488 @@ public async Task DevicesClient_DeleteAsync_OnlyIfUnchangedTrueHasEtags()
+            var mockHttpClient = new Mock<HttpClient>();
+
+            var devicesClient = new DevicesClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory.Object,
+                s_retryHandler);
+
+            // act
+            Func<Task> act = async () => await devicesClient.GetJobAsync(null).ConfigureAwait(false);
+
+            // assert
+            await act.Should().ThrowAsync<ArgumentNullException>().ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task DevicesClient_GetJobAsync_EmptyJobIdThrows()
+        {
+            // arrange
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            var mockHttpRequestFactory = new Mock<HttpRequestMessageFactory>();
+            var mockHttpClient = new Mock<HttpClient>();
+
+            var devicesClient = new DevicesClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory.Object,
+                s_retryHandler);
+
+            // act
+            Func<Task> act = async () => await devicesClient.GetJobAsync("").ConfigureAwait(false);
+
+            // assert
+            await act.Should().ThrowAsync<ArgumentException>().ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task DevicesClient_GetJobAsync_InvalidJobIdThrows()
+        {
+            // arrange
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            var mockHttpRequestFactory = new Mock<HttpRequestMessageFactory>();
+            var mockHttpClient = new Mock<HttpClient>();
+
+            var devicesClient = new DevicesClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory.Object,
+                s_retryHandler);
+
+            // act
+            Func<Task> act = async () => await devicesClient.GetJobAsync("sample job").ConfigureAwait(false);
+
+            // assert
+            await act.Should().ThrowAsync<ArgumentException>().ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task DevicesClient_GetJobsAsync()
+        {
+            // arrange
+            var jobId = "sampleJob";
+            var jobStatus = JobStatus.Completed;
+
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            mockCredentialProvider
+                .Setup(getCredential => getCredential.GetAuthorizationHeader())
+                .Returns(s_validMockAuthenticationHeaderValue);
+            var mockHttpRequestFactory = new HttpRequestMessageFactory(s_httpUri, "");
+            var jobsToReturn = new List<IotHubJobResponse>() { new IotHubJobResponse() { JobId = jobId, Status = jobStatus } };
```

Comment:
You can drop the ctor parens when using an initializer list.

```suggestion
            var jobsToReturn = new List<IotHubJobResponse> { new IotHubJobResponse { JobId = jobId, Status = jobStatus } };
```